csrc/moe/marlin_moe_ops.cu

```diff
@@ -1749,6 +1749,9 @@ void marlin_mm_moe_f16i4(const void* A, const void* B, void* C,
                                 has_act_order, is_k_full, max_shared_mem);
   }
 
+  int group_tensor_size =
+      (!is_k_full && has_act_order) ? prob_k / num_groups : group_size;
+
```

Comment:
A couple of nits here:
* Could you add comment describing what this is and why we need to handle it differently for the `!is_k_full` case?
* Doesn't `!is_k_full` imply `has_act_order`?
https://github.com/vllm-project/vllm/blob/5f7bb584272ee15147a411b887e7ababd6b9b9d0/vllm/model_executor/layers/quantization/utils/marlin_utils.py#L147-L148

* Since `group_tensor_size` is only used in one place, I think it would improve readability to define it right where it's used.

Reply:
2. It does. I wrote it like this to keep it consistent with the condition in the first line of `get_scales_cache_size()`. I will add a relevant assert instead.

Reply:
actually, this variable can be completely removed, `num_groups` is the same thing